src/metalog.c

```diff
@@ -459,6 +471,53 @@ static int configParser(const char * const file)
             break;
         }
     }
+
+    /* read all config files of an eventually configured directory */
+    if (config_dir != NULL) {
+        FILE *fp2;
+        char *p = NULL;
+        DIR *dp;
+        struct dirent *ep;
+
+        dp = opendir(config_dir);
+        if (dp == NULL) {
+            retcode = -1;
+            goto rtn;
+        }
+
+        /* config file names must end with ".conf" */
+        while ((ep = readdir(dp)) != NULL) {
+            p = strchr(ep->d_name, (int) '.');
+            if (p == NULL || !strstr(p, ".conf")) {
+                continue;
+            }
+
+            if ((file_path = wmalloc(strlen(ep->d_name) + strlen(config_dir) + 2)) == NULL) {
+                retcode = -1;
+                goto rtn;
+            }
+            sprintf(file_path, "%s/%s", config_dir, ep->d_name);
+            if ((fp2 = fopen(file_path, "r")) == NULL) {
+                warnp("Can't open the config file %s", file_path);
+                retcode = -1;
+                goto rtn;
+            }
+
+            while (fgets(line, sizeof line, fp2) != NULL) {
+                if ((retcode = parseLine(line, &cur_block, &default_block,
+                                         re_newblock, re_newstmt,
+                                         re_comment, re_null)) != 0) {
+                    break;
+                }
+            }
+
+            free(file_path);
+            file_path = NULL;
```

Comment:
Why not declare `file_path` inside the while loop so it doesn't need to be nulled here?

Reply:
This was a left over from another try. I'm not very happy with this goto at all. And there was one free() missing before the "goto".